pkg/storage/etcd/etcd_helper.go

```diff
@@ -40,15 +41,33 @@ import (
 	etcdutil "k8s.io/kubernetes/pkg/storage/etcd/util"
 )
 
+// ValueTransformer allows a string value to be transformed before being read from or written to the underlying store. The methods
+// must be able to undo the transformation caused by the other.
+type ValueTransformer interface {
+	// TransformStringFromStorage may transform the provided string from its underlying storage representation or return an error.
+	TransformStringFromStorage(string) (string, error)
```

Comment:
I want to ensure that one implementation can satisfy both interfaces (a decrypter) so the names need to be different.